README.md

```diff
@@ -44,10 +45,162 @@ thrift_files: Mix.Utils.extract_files(["thrift"], [:thrift])
 By default, the generated source files will be written to the `lib` directory,
 but you can change that using the `thrift_output` option.
 
+## Working with Thrift
+
+The following examples use the following thrift definition:
+
+```thrift
+namespace elixir Thrift.Test 
+exception UserNotFound {
+  1: string message
+}
+struct User {
+  1: i64 id,
+  2: string username,
+  3: string first_name,
+  4: string last_name
+}
+
+service UserService {
+  1: bool ping(),
+  2: User get_user_by_id(1: i64 user_id) throws (1: UserNotFound unf),
+  3: boolean deleteUser(1: i64 userId),
+}
+```
+
+#### The generated code will be placed in the following modules:
+
+Generated Code | Output Module
+---------------|----------------
+User Struct    | `Thrift.Test.User`
+UserNotFound Exception | `Thrift.Test.UserNotFound`
+User Binary Protocol | `Thrift.Test.User.BinaryProtocol`
+UserNotFound Binary Protocol | `Thrift.Test.UserNotFound.BinaryProtocol`
+UserService Framed Binary Client | `Thrift.Test.UserService.Clients.Binary.Framed`
+UserService Framed Binary Server | `Thrift.Test.UserService.Servers.Binary.Framed`
+UserService Handler Behviour (Used for writing servers)    | `Thrift.Test.UserService.Handler.Behaviour`
+
+
+## Using the Client 
+The client includes a static module that does most of the work, and a generated interface module that performs some conversions and makes calling remote functions easier. You will not directly interface with the static module, but it is the one that's started when `start_link` is called. 
+The static client module uses [James Fish](http://github.com/fishcakez)'s excellent [connection](http://github.com/fishcakez/connection) behaviour. 
+
+For each function defined in the service, the generated module has four functions. Consider the `get_user_by_id` function above.
+
+Function name  | Description 
+---------------|----------
+`get_user_by_id/2` | Makes a request to the remote `get_user_by_id` RPC. Returns `{:ok, response}` or `{:error, reason}` tuples.
+`get_user_by_id!/2`  | Same as above, but raises an exception if something goes wrong. The type of exception can be one of the exceptions defined in the service or `Thrift.TApplicationException`.
+`get_user_by_id_with_options/3` | Allows you to pass `gen_tcp` and `GenServer` options to your client. This is useful for setting the `GenServer` timeout if you expect your RPC to take longer than the default of 5 seconds. Like `get_user_by_id/2`, this function returns `{:ok, response}` or `{:error, reason}` tuples.
+`get_user_by_id_with_options!/3` | Allows you to pass `gen_tcp` and `GenServer` options and raises an exception if an error occurs. 
+
+#### Note: in the above example, the function `deleteUser` will be converted to `delete_user` to comply with Elixr's naming conventions.
```

Comment:
Make `Note` be bold and the rest of the sentence normal emphasis?  It would be nice to provide a link to the naming conventions.  As usual, Elixir has a fantastic set of documentation for this: http://elixir-lang.org/docs/stable/elixir/naming-conventions.html